src/runtime/HalidePyTorchHelpers.h

```diff
@@ -0,0 +1,91 @@
+#ifndef HL_PYTORCH_WRAPPER_H
+#define HL_PYTORCH_WRAPPER_H
+
+/** \file
+ * Set of utility functions to wrap PyTorch tensors into Halide buffers,
+ * making sure the data in on the correct device (CPU/GPU).
+ */
+
+#include <exception>
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <vector>
+
+#include "torch/extension.h"
+
+#include "HalideBuffer.h"
+
+#ifdef HL_PT_CUDA
+#include "HalideRuntimeCuda.h"
+#include "cuda.h"
+#include "cuda_runtime.h"
+#endif
+
+#define WEAK __attribute__((weak))
+
+#define HLPT_CHECK_CONTIGUOUS(x) AT_ASSERTM(x.is_contiguous(), #x " must be contiguous")
+#define HLPT_CHECK_CUDA(x) AT_ASSERTM(x.type().is_cuda(), #x " must be a CUDA tensor")
+#define HLPT_CHECK_DEVICE(x, dev) AT_ASSERTM(x.is_cuda() && x.get_device() == dev, #x " must be a CUDA tensor")
+
+
+namespace Halide {
+namespace PyTorch {
+
+using Halide::Runtime::Buffer;
+
+inline std::vector<int> get_dims(const at::Tensor tensor) {
```

Comment:
four-space indents for all C++ code in Halide, please

Reply:
done

Reply:
ok